src/components/views/dialogs/CreateRoomDialog.js

```diff
@@ -19,55 +19,172 @@ import createReactClass from 'create-react-class';
 import PropTypes from 'prop-types';
 import sdk from '../../../index';
 import SdkConfig from '../../../SdkConfig';
+import withValidation from '../elements/Validation';
 import { _t } from '../../../languageHandler';
+import MatrixClientPeg from '../../../MatrixClientPeg';
 
 export default createReactClass({
     displayName: 'CreateRoomDialog',
     propTypes: {
         onFinished: PropTypes.func.isRequired,
     },
 
-    componentWillMount: function() {
+    getInitialState() {
         const config = SdkConfig.get();
-        // Dialog shows inverse of m.federate (noFederate) strict false check to skip undefined check (default = true)
-        this.defaultNoFederate = config.default_federate === false;
+        return {
+            isPublic: false,
+            name: "",
+            topic: "",
+            alias: "",
+            detailsOpen: false,
+            noFederate: config.default_federate === false,
+            nameIsValid: false,
+        };
     },
 
-    onOk: function() {
-        this.props.onFinished(true, this.refs.textinput.value, this.refs.checkbox.checked);
+    _roomCreateOptions() {
+        const createOpts = {};
+        createOpts.name = this.state.name;
+        if (this.state.isPublic) {
+            createOpts.visibility = "public";
+            createOpts.preset = "public_chat";
+            // to prevent createRoom from enabling guest access
+            createOpts['initial_state'] = [];
+            const {alias} = this.state;
+            const localPart = alias.substr(1, alias.indexOf(":") - 1);
+            createOpts['room_alias_name'] = localPart;
+        }
+        if (this.state.topic) {
+            createOpts.topic = this.state.topic;
+        }
+        if (this.state.noFederate) {
+            createOpts.creation_content = {'m.federate': false};
+        }
+        return createOpts;
+    },
+
+    componentDidMount() {
+        this._detailsRef.addEventListener("toggle", this.onDetailsToggled);
```

Comment:
I assume an `onToggle` react listener doesn't work for this?

Reply:
Indeed